### PR TITLE
[LLM] Fix llm models extension issue

### PR DESCRIPTION
**Pull request by changwangss**
## Type of Change

Fix llm models extension test issue
1. phi transformers version.
2. add requirements for baichuan/qwen/codegen.

Support codegen sq workflow, due to codegen use the remote tokenizer, so add `trust_remote_code` to get the correct tokenizer. transformers version requests 4.35.2.
fp32 with limit 20
```
numactl -m 0 -C 0-55 python run_generation.py     --model Salesforce/codegen25-7b-multi    --accuracy     --batch_size 20     --n_samples 20     --allow_code_execution     --temperature 0.2     --do_sample     --tasks "humaneval" --trust_remote_code True
```
```
Evaluating generations...
{
  "humaneval": {
    "pass@1": 0.5725,
    "pass@10": 0.649981867977224
  },
  "config": {
    "model": "llama",
    "temperature": 0.2,
    "n_samples": 20
  }
}
{'humaneval': {'pass@1': 0.5725, 'pass@10': 0.649981867977224}, 'config': {'model': 'llama', 'temperature': 0.2, 'n_samples': 20}}

```
int8 sq with alpha 0,5 calib_iters 5 limit 20
```
python run_generation.py     --model Salesforce/codegen25-7b-multi        --alpha 0.5       --accuracy     --batch_size 20     --n_samples 20     --allow_code_execution     --temperature 0.2     --do_sample     --tasks "humaneval" --limit 20 --trust_remote_code True --sq --int8 --calib_iters 5
```
```
Evaluating generations...
{
  "humaneval": {
    "pass@1": 0.3575,
    "pass@10": 0.4497261252679209
  },
  "config": {
    "model": "llama",
    "temperature": 0.2,
    "n_samples": 20
  }
}
{'humaneval': {'pass@1': 0.3575, 'pass@10': 0.4497261252679209}, 'config': {'model': 'llama', 'temperature': 0.2, 'n_samples': 20}}

```
## Description

detail description 
JIRA ticket: xxx

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Comment by VincyZhang** [comment]
https://inteltf-jenk.sh.intel.com/view/ITREX-release/job/ITREX-1.3-release-test/44/

**Comment by changwangss** [comment]
https://inteltf-jenk.sh.intel.com/view/ITREX-release/job/ITREX-1.3-release-test/47/

**Review by thuang6 — APPROVED** [review]

**Comment by VincyZhang** [comment]
https://inteltf-jenk.sh.intel.com/job/ITREX-1.3-release-test/48/

**Review by VincyZhang — APPROVED** [review]

File: examples/huggingface/pytorch/code-generation/quantization/run_generation.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "--model", nargs="?", default="bigcode/starcoderbase", const="bigcode/starcoderbase"
 )
 parser.add_argument("--trust_remote_code", default=False)
-parser.add_argument("--revision", default="main", type=str)
```
Comment (hshen14) [inline examples/huggingface/pytorch/code-generation/quantization/run_generation.py:31]: why changing the parameter? it will introduce incompatible change.

Comment (changwangss) [reply examples/huggingface/pytorch/code-generation/quantization/run_generation.py:31]: _commit_hash can work, revision can used to get the correct config, but for model_class it failed. it is not incompatible change, I only add _commit_hash and doesn't delete revision.

File: intel_extension_for_transformers/transformers/utils/utility.py [hunk 2]
```diff
@@ -196,7 +196,7 @@ def generate_dummy_past_key_values_for_opt_llm(config, input_bs, num_beams=1):
     "llama",
     "mistral",
     "chatglm",
-    "falcon"
+    "qwen"
```
Comment (VincyZhang) [inline intel_extension_for_transformers/transformers/utils/utility.py:199]: falcon can be added back?

**Comment by changwangss** [comment]
local test qwen
image

**Comment by VincyZhang** [comment]
CI all pass, ready for merge.

**Pull request merged**